cpp/src/parquet/printer.cc

```diff
@@ -64,6 +64,25 @@ void PrintPageEncodingStats(std::ostream& stream,
 // the fixed initial size is just for an example
 #define COL_WIDTH 30
 
+void Put(std::ostream& stream, char c, int n) {
+  for (int i = 0; i < n; ++i) {
+    stream.put(c);
+  }
+}
+
+void PrintKeyValueMetadata(std::ostream& stream,
+                           const KeyValueMetadata& key_value_metadata,
+                           int indent_level = 0, int indent_width = 1) {
```

Comment:
Seems like it would be easier to just pass the indentation string?
```suggestion
                           std::string_view indent = "") {
```

Reply:
To mimic the original output, PrintKeyValueMetadata needs to print two different levels of indentation.